lib/matplotlib/cbook/__init__.py

```diff
@@ -1617,17 +1645,17 @@ def normalize_kwargs(kw, alias_mapping=None, required=(), forbidden=(),
         If an Artist subclass or instance is passed, use its properties alias
         mapping.
 
-    required : iterable, optional
-        A tuple of fields that must be in kwargs.
+    required : list of str, optional
```

Comment:
These don't have to be lists, any non-generator iterable will work.

Reply:
I would leave that. `non-generator iterable of str` feels a bit too bulky. The essence that it's a homogeneous sequence of str is best described by `list of str`. I find it ok to accept a slightly broader range of types than documented in this case.

Note: I searched through the code to find out how it's actually used and found out that neither of `required`, `forbidden` and `allowed` are used anywhere.

Reply:
I agree that list is the "practical" documentation.
Unless we have concrete plans to use `required`/`forbidden`/`allowed` I would suggest just deprecating them.  IIRC @tacaswell you wrote this helper originally; thoughts?